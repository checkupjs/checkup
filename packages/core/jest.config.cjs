module.exports = {
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'json'],
  transform: { '\\.ts$': 'ts-jest' },
  coverageReporters: ['lcov', 'text-summary'],
  // collectCoverage: !!process.env.CI,
  collectCoverageFrom: ['src/**/*.ts'],
  coveragePathIgnorePatterns: ['/templates/'],
  coverageThreshold: {
    global: {
      branches: 100,
      functions: 100,
      lines: 100,
      statements: 100,
    },
  },
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],
  snapshotFormat: {
    printBasicPrototype: false,
  },
  testPathIgnorePatterns: ['/__fixtures__/', '/__utils__/'],
  extensionsToTreatAsEsm: ['.ts'],
  globals: {
    'ts-jest': {
      useESM: true,
    },
  },
  moduleNameMapper: {
    '^(\\.{1,2}/.*)\\.js$': '$1',
  },
};
