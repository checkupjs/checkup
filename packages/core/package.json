{
  "name": "@checkup/core",
  "description": "Checkup's core library",
  "version": "0.0.0",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {},
  "devDependencies": {
    "@types/chai": "^4",
    "@types/jest": "^25.1.1",
    "@types/node": "^13",
    "chai": "^4.2.0",
    "jest": "^25.1.0",
    "stdout-stderr": "^0.1.13",
    "ts-jest": "^25.2.0",
    "ts-node": "^8",
    "type-fest": "^0.10.0",
    "typescript": "^3.3"
  },
  "engines": {
    "node": "10.* || >= 12.*"
  },
  "files": [
    "/lib"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc",
    "build:watch": "yarn build -w",
    "test": "jest"
  },
  "types": "lib/index.d.ts"
}
