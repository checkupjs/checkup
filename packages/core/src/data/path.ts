export function trimCwd(path: string, cwd: string) {
  return path.replace(`${cwd}/`, '');
}

export function trimAllCwd(paths: string[], cwd: string) {
  return paths.map((path) => trimCwd(path, cwd));
}
