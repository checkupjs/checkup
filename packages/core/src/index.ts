export { default as BaseTask } from './base-task';
export { default as BaseTaskResult } from './base-task-result';
export { default as FileSearcherTask } from './file-searcher-task';
export { default as FileSearcher } from './searchers/file-searcher';
export { default as CardData } from './pdf-components/card-data';
export { loadPlugins } from './loaders/plugin-loader';
export { getPackageJson } from './utils/get-package-json';
export { default as CheckupConfigService } from './configuration/checkup-config-service';
export { default as getSearchLoader } from './configuration/loaders/get-search-loader';
export { default as getFilepathLoader } from './configuration/loaders/get-filepath-loader';
export { ui } from './utils/ui';
export { toPairs } from './utils/data-transformers';

export * from './types';
