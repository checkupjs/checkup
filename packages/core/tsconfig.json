{
  "compilerOptions": {
    "declaration": true,
    "composite": true,
    "importHelpers": true,
    "module": "commonjs",
    "outDir": "lib",
    "rootDir": "src",
    "strict": true,
    "target": "es2017",
    "types": ["chai", "jest", "node"]
  },
  "include": ["src/**/*"]
}
