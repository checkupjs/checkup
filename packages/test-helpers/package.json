{
  "name": "@checkup/test-helpers",
  "description": "Checkup's test helper library",
  "version": "0.0.10",
  "author": "Steve Calvert <steve.calvert@gmail.com>",
  "bugs": "https://github.com/checkupjs/checkup/issues",
  "dependencies": {
    "@checkup/core": "0.0.10",
    "fixturify-project": "^2.1.0",
    "handlebars": "^4.7.6",
    "stdout-stderr": "^0.1.13",
    "tmp": "^0.2.0"
  },
  "engines": {
    "node": ">= 12.*"
  },
  "files": [
    "/lib",
    "/static"
  ],
  "homepage": "https://github.com/checkupjs/checkup",
  "license": "MIT",
  "main": "lib/index.js",
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/checkupjs/checkup",
  "scripts": {
    "build": "tsc --build",
    "test": "jest --passWithNoTests --no-cache"
  },
  "types": "lib/index.d.ts"
}
