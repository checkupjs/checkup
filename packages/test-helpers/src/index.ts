export { default as Project } from './ember-cli-fixturify-project';
export { stdout } from './stdout';
export { default as CheckupProject } from './checkup-fixturify-project';
export { default as Plugin } from './plugin';
