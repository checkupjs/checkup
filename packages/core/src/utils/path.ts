import { isAbsolute, join, relative } from 'path';

export function toAbsolute(baseDir: string, filePath: string) {
  let absolutePath = filePath;

  if (!isAbsolute(filePath)) {
    absolutePath = join(baseDir, filePath);
  }

  return absolutePath;
}

export function toRelative(baseDir: string, filePath: string) {
  let relativePath = filePath;

  if (isAbsolute(relativePath)) {
    relativePath = relative(baseDir, filePath);
  }

  return relativePath;
}
