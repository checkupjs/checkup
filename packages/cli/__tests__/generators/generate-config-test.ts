import * as helpers from 'yeoman-test';
import * as fs from 'fs';
import * as path from 'path';
import { CheckupConfigFormat } from '@checkup/core';
import ConfigGenerator from '../../src/generators/config';
import { testRoot } from '@checkup/test-helpers';

describe('config-init-generator', () => {
  it('should write a config given default answers', async () => {
    const directory = await helpers.run(ConfigGenerator).withPrompts({
      framework: 'Other',
      format: CheckupConfigFormat.JSON,
    });

    expect(testRoot(directory).file('.checkuprc').contents).toMatchSnapshot();
  });

  it('should write a config given Ember framework', async () => {
    const directory = await helpers.run(ConfigGenerator).withPrompts({
      framework: 'Ember',
      format: CheckupConfigFormat.JSON,
    });

    expect(testRoot(directory).file('.checkuprc').contents).toMatchSnapshot();
  });

  it('should error if a checkuprc file is already present', async () => {
    await expect(
      helpers.run(ConfigGenerator).inTmpDir(function (dir) {
        fs.writeFileSync(path.join(dir, '.checkuprc'), JSON.stringify({}));
      })
    ).rejects.toThrow();
  });
});
