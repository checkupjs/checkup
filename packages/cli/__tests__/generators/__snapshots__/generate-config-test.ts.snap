// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-init-generator should write a config given Ember framework 1`] = `
"{
  \\"plugins\\": [
    \\"@checkup/plugin-ember\\"
  ],
  \\"tasks\\": {}
}"
`;

exports[`config-init-generator should write a config given default answers 1`] = `
"{
  \\"plugins\\": [],
  \\"tasks\\": {}
}"
`;
